tools/clang/test/CodeGenDXIL/hlsl/intrinsics/maybereorder.hlsl

```diff
@@ -0,0 +1,37 @@
+// RUN: %dxc -T lib_6_9 -E main %s | FileCheck %s --check-prefix DXIL
+// RUN: %dxc -T lib_6_9 -E main %s -fcgl | FileCheck %s --check-prefix FCGL
+// RUN: %dxc -T lib_6_9 -E main %s -ast-dump-implicit | FileCheck %s --check-prefix AST
```

Comment:
I think AST tests belong under SemaHLSL.  We could probably address this later for these tests, but in the future, we should be putting them there.  For -fcgl and full compilation, that should go here, but for tests not under a directory already excluded from unsupported DXIL versions, we need a `// REQUIRES: dxil-1-9` in the test.

Reply:
Tracking this in #7359